### ISSUE TITLE
Create a long-living graphical window

### ISSUE THREAD

**Issue by hintron** [hintron/computer-enhance#39]
The problem I am having right now is that when I display a frame to memory, I create an event loop and `winit` window ad-hoc. So I can't  render the next frame without leaving the loop. Somehow I need to create this window in a separate thread, and then have it live between instruction iterations so I can render to it.

I think the simplest way to do this is to create a graphical window on execution startup, and have it waiting around until we need it. Another alternative is to create it ad-hoc when we need it, and somehow save off the pointers to it and use it on the next iteration (though I'm not sure how easy that will be in Rust - it should be doable with a move, but who knows).

If I can get a general purpose graphical output working, then I can split it up so that some of it is the true program's output, some of it is a memory mini-heatmap, and some of it is register and instruction visualizations. Then I can eventually turn this emulator into a true debugger, like emu86.

**Comment by hintron** [comment]
winit does not like it when I try to create the event loop outside of the main thread:

> thread '\<unnamed\>' panicked at /home/hintron/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.14/src/platform_impl/linux/mod.rs:747:13: Initializing the event loop outside of the main thread is a significant cross-platform compatibility hazard. If you absolutely need to create an EventLoop on a different thread, you can use the `EventLoopBuilderExtUnix::any_thread` function.

**Comment by hintron** [comment]
I was hoping to have the graphical window be in its own thread, and then I would send data to it to display, but that might not be possible. Or, perhaps I should look into what the cross-platform compatibility hazards are and determine if it's ok (looks like it's MacOS - it requires all UI-related things to be called in the main thread).

**Comment by hintron** [comment]
If I can't make the graphical output a separate thread, either I have to 1) merge the graphical thread with everything else or 2) make everything else run in a separate thread. I think 2) would be the easiest path forward.

Once I have two threads, I think I want to simply pass a immutable reference to a frame buffer slice from the simulation thread to the graphical thread. Then, the graphical thread can render it, and when it's done, return control back to the simulation thread to keep running.

I could also make a copy of the frame buffer, so that rendering does not hold back simulation. I could also implement a double-buffer system to have maximum performance, so that the graphical system is always working on a frame buffer.

I also want to try to limit frames per second to something like 60 in the simulator, so that I can tweak the game to work with that timing. I don't want it to zoom too fast when I make a good optimization.

**Comment by hintron** [comment]
2) works (keep graphics in main thread, move all emulator logic into separate thread).

Currently, I'm making copies of the images to display and sending it from the emulator to the graphics thread through a mpsc channel. It's simple and it works, so I will keep using that until I need to optimize.

**Comment by hintron** [comment]
This has been implemented in commits e0b486c...0b12958. Notably:
* a4805182740ed5c1060c9b3ebb20b82916aec3ff changes the winit display loop to do polling instead of waiting
* c5996ddf2dc71c766213710e37104fbc859cd9dd makes the initial display size 1024x1024
* Created a new splash screen image in c741b2785b0d85326a1667bb02fbd76f25917432
* ca58eca447826cd890638a404e1147bc46fe9ad4 implements the new graphical loop. This is the main commit. I pass images to render from the emulation thread to the main thread via mpsc channels. The graphics thread needed to be in the main thread to avoid issues, as mentioned above.
* Fixed the issue of inverse blue and red channels in b79fce0518cbb47a31d4fe9aae5c9ed15d7400c9

I am still having quite a bit of issues with WSL2 displaying images, but when I tried on native Linux, things worked smoothly. So I will investigate those issues in another issue. For now, it
